### PR TITLE
Prac 5 code review request

### PR DESCRIPTION
**Pull request by Salvin-George**
Hey, can you review this code for me. Specifically i want to improve on my use of EAFP approach and incorporate it more into the error checking of my code :). @RMabo

### CODE REVIEW

File: prac_05/state_names.py
Comment (RMabo) [inline prac_05/state_names.py:1]: Great code @Salvin-George. Maybe consider breaking your code into functions for better readability.

File: prac_05/emails.py
Comment (RMabo) [inline prac_05/emails.py:1]: Great job on this code! 

I see you added a break - maybe consider replacing it with the if-else statement. 
 while email != '':
        name = extract_name(email)
        confirm = input(f"Is this your name, {name}? (Y/n)").strip().lower()
        if confirm != "n":
            name_to_email[email] = name
        else:
            user_name = input("Name: ")
            name_to_email[email] = user_name. 
hope this help :)

File: prac_05/hex_colours.py
Comment (RMabo) [inline prac_05/hex_colours.py:1]: Great code! :)

File: prac_05/word_occurrences.py
Comment (RMabo) [inline prac_05/word_occurrences.py:1]: Great code! Maybe break it into a function

File: prac_05/wimbledon.py
Comment (RMabo) [inline prac_05/wimbledon.py:1]: Great stuff! Don't forget docstrings to explain what each function is doing. 

You also have have  read_file() but forgot to add in_file.close()

Overall, this is a great piece of code. :)

File: prac_05/emails.py
Comment (Salvin-George) [reply prac_05/emails.py:1]: yea thx :D

**Pull request merged**